alf/algorithms/dsac_algorithm.py

```diff
@@ -266,39 +263,22 @@ def _compute_critics(self,
 
     def _critic_train_step(self, inputs: TimeStep, state: SacCriticState,
                            rollout_info: SacInfo, action, tau_info: TauInfo):
-        bs = self._mini_batch_size
         # Calculate the critics and value for both the current observation
         # and next observation
-        # [B * (T-1), n_quantiles]
-        critic_observation = nest.map_structure(lambda tensor: tensor[:-bs],
-                                                inputs.observation)
-        critic_action = nest.map_structure(lambda tensor: tensor[:-bs],
-                                           rollout_info.action)
+        # [B * T, n_quantiles] or [B, n_quantiles]
         critics, critics_state, _ = self._compute_critics(
             self._critic_networks,
-            critic_observation,
-            critic_action,
-            tau_info.tau_hat[:-bs],
+            inputs.observation,
+            rollout_info.action,
+            tau_info.tau_hat,
             state.critics,
             replica_min=False)
-        # Extend with zeros so that the tensors can be correctly reshape to
-        # [T, B, ...]
-        zeros = torch.zeros_like(critics[:bs])
-        critics = torch.cat((critics, zeros), dim=0)
 
         with torch.no_grad():
-            target_critic_observation = nest.map_structure(
-                lambda tensor: tensor[bs:], inputs.observation)
-            target_critic_action = nest.map_structure(
-                lambda tensor: tensor[bs:], action)
             target_critics, target_critics_state, _ = self._compute_critics(
-                self._target_critic_networks, target_critic_observation,
-                target_critic_action, tau_info.next_tau_hat[bs:],
-                state.target_critics, tau_info.next_delta_tau[bs:])
-            # Prepend with zeros so that the tensors can be correctly reshape to
-            # [T, B, ...]
-            zeros = torch.zeros_like(target_critics[:bs])
-            target_critics = torch.cat((zeros, target_critics), dim=0)
+                self._target_critic_networks, inputs.observation, action,
+                tau_info.next_tau_hat, state.target_critics,
+                tau_info.next_delta_tau)
```

Comment:
what's the reason for changes?

Reply:
Similar as above, previously, both ``inputs.observation`` and ``action`` are chunked to extract the last [B * (T-1), ...] tensors to forward the critics, instead of forwarding the full tensors of shape [B * T, ...]. Now change it back.